pkg/api/server/v1alpha2/result/result.go

```diff
@@ -73,3 +75,12 @@ func Match(r *pb.Result, prg cel.Program) (bool, error) {
 	}
 	return resultscel.Match(prg, "result", r)
 }
+
+// GetResultByParentName is a helper function to retrieve a Result from database by parent and name
+func GetResultByParentName(gdb *gorm.DB, parent, name string) (*db.Result, error) {
```

Comment:
I might have typo-ed a previous comment - I think pkg/api/server/v1alpha2/results.go is probably the best spot for this, since this where the other DB operations are.

We should also unexport this so that this isn't publicly visible.